pkg/cmd/checkendpoints/controller/connection_checker.go

```diff
@@ -233,16 +233,30 @@ func manageStatusOutage(recorder events.Recorder) v1alpha1helpers.UpdateStatusFu
 		if len(status.Successes) > 0 {
 			latestSuccess = status.Successes[0]
 		}
-		if currentOutage == nil {
-			if latestFailure.Start.After(latestSuccess.Start.Time) {
-				recorder.Warningf("ConnectivityOutageDetected", "Connectivity outage detected: %s", latestFailure.Message)
-				status.Outages = append([]operatorcontrolplanev1alpha1.OutageEntry{{Start: latestFailure.Start}}, status.Outages...)
+		switch {
+		case currentOutage == nil && latestFailure.Start.After(latestSuccess.Start.Time):
+			// outage started
+			newOutage := operatorcontrolplanev1alpha1.OutageEntry{
+				Start:     latestFailure.Start,
+				StartLogs: []operatorcontrolplanev1alpha1.LogEntry{latestFailure},
+				Message:   fmt.Sprintf("Connectivity outage detected at %v", latestFailure.Start.Format(time.RFC3339Nano)),
 			}
-		} else {
-			if latestSuccess.Start.After(latestFailure.Start.Time) {
-				currentOutage.End = latestSuccess.Start
-				recorder.Eventf("ConnectivityRestored", "Connectivity restored after %v: %s", currentOutage.End.Sub(currentOutage.Start.Time), latestSuccess.Message)
+			status.Outages = append([]operatorcontrolplanev1alpha1.OutageEntry{newOutage}, status.Outages...)
+			recorder.Warningf("ConnectivityOutageDetected", "Connectivity outage detected: %s", latestFailure.Message)
+		case currentOutage != nil && latestFailure.Start.After(latestSuccess.Start.Time):
+			// outage ongoing
+			if len(currentOutage.StartLogs) == 0 || currentOutage.StartLogs[0].Message != latestFailure.Message {
+				currentOutage.StartLogs = append([]operatorcontrolplanev1alpha1.LogEntry{latestFailure}, currentOutage.StartLogs...)
```

Comment:
this will produce infinite startlogs, right?  We want a cap.

Also, we want `EndLogs` to always have the last five-ish failures.

Reply:
Done.